### DISCUSSION TITLE
The translation doesnt work if I use the browser back and forward buttons

### DISCUSSION THREAD

**Discussion by aswinfrancis** [i18next/i18next#1592] [Q&A]
The translation works fine during normal navigation but if I am on a page that uses i18next translate and then I hit the browser back button and then again forward button, the page would show the key and not the translated text. If I refresh the page, the keys load again.

Any idea how I can sort this? FYI, I am not using  promise to check if the init is done or not. I load the necessary resources early on and hence initialization is done by the time I start calling t(). Not sure if its related but this is not causing any problem when navigation to any of the countless pages we have via the site.

Another unrelated question I have is why "en" is added to language hierarchy even though I have set the fallback language to 'en-CA'?

**Comment by jamuhl** [comment]
looks like a problem only you can solve - must be something in your pages
